src/v/raft/append_entries_buffer.cc

```diff
@@ -40,9 +40,20 @@ append_entries_buffer::enqueue(append_entries_request&& r) {
 }
 
 ss::future<> append_entries_buffer::stop() {
+    auto f = _gate.close();
     _enqueued.broken();
     _flushed.broken();
-    return _gate.close();
+    auto response_promises = std::exchange(_responses, {});
+    // set errors
+    for (auto& p : response_promises) {
+        p.set_exception(ss::gate_closed_exception());
+    }
+    vassert(
+      _responses.empty(),
```

Comment:
after this runs above `auto response_promises = std::exchange(_responses, {});` this assertion would never fail since there is no preemption scheduling preemption in this routine. is there a stronger test?

Reply:
this is enough to state that there will be no promises pending to be set.

Reply:
noah is right. we can just move this as a comment. since _responses are empty.